compilerla/data-donuts
# Chore: devcontainer

Closes #93

## What this PR does

Introduces a devcontainer setup, to make it easier to build and run this site locally. Devcontainer installs dependencies as root, but switches to a non-root user `compiler` 

❗❗ Reorganizes the site content (e.g. HTML, CSS, images) into the `src/` directory to keep things together and follow our normal pattern in other sites.

Added this repo as a Netlify site so we get free PR / branch previews. Will turn off GitHub pages once this PR is merged, and migrate DNS to the Netlify site #112 

## Testing this PR

1. Checkout the branch
2. Open in VS Code
3. Rebuild and Reopen in devcontainer
4. Run the build task `> Tasks: Run Build Task` or `Ctrl/Cmd + Shift + B`
5. Confirm the site builds and you can open it at `http://localhost:PORT` (`4000` by default but assigned by Docker)
6. Confirm no visual or content regressions

## See also 

compilerla/compiler.la#123
